mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Energy32.java

```diff
@@ -0,0 +1,53 @@
+package no.nordicsemi.android.mesh.sensorutils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RestrictTo;
+
+/**
+ * The Energy 32 characteristic is used to represent a measure of energy in units of kilowatt-hours.
+ */
+public class Energy32 extends DevicePropertyCharacteristic<Double> {
+
+    private static final String TAG = Count.class.getSimpleName();
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY)
+    public Energy32(@NonNull final byte[] data, final int offset, final int length) {
+        super(data, offset, length);
+        long bits = 0;
+        for (byte b : data) bits = (bits >> 8) + (((long) b & 0xFF) << 24);
```

Comment:
Interesting to see + instead of | (or) when doing bit compilation like this.

Reply:
Agree, should be | for less cognitive load.